.github/workflows/test-suite.yml

```diff
@@ -2,24 +2,47 @@ name: 'Test Suite'
 
 on:
   workflow_dispatch:
+    inputs:
+      terraform_matrix:
+        description: 'Terraform version matrix (JSON array)'
+        default: '["1.6.x", "1.5.x", "1.4.x", "1.3.x", "1.2.x"]'
+      provider_matrix:
+        description: 'MongoDB Atlas Provider version matrix (JSON array)'
+        default: '["", "1.12.3"]' # "" for latest version
+
   schedule:
     - cron: "0 0 * * *" # workflow runs every day at midnight UTC
-  
+   
+concurrency:
+  group: '${{ github.workflow }}'
+  cancel-in-progress: false
```

Comment:
only one Test Suite run at a time to avoid interference. new runs will wait until previous ones finish